transport/packet_test.go

```diff
@@ -69,6 +70,28 @@ func TestUDPEndpointDomain(t *testing.T) {
 	assert.Equal(t, resolvedAddr, conn.RemoteAddr().String())
 }
 
+func TestFuncPacketEndpoint(t *testing.T) {
+	expectedConn := &fakeConn{}
+	expectedErr := errors.New("fake error")
+	endpoint := FuncPacketEndpoint(func(ctx context.Context) (net.Conn, error) {
+		return expectedConn, expectedErr
+	})
+	conn, err := endpoint.Connect(context.Background())
+	require.Equal(t, expectedConn, conn)
+	require.Equal(t, expectedErr, err)
+}
+
+func TestFuncPacketDialer(t *testing.T) {
+	expectedConn := &fakeConn{}
+	expectedErr := errors.New("fake error")
+	dialer := FuncPacketDialer(func(ctx context.Context, add string) (net.Conn, error) {
+		return expectedConn, expectedErr
```

Comment:
Maybe good to assert the parameters are expected as well?

```suggestion
		require.Equal(t, "unused", addr)
		return expectedConn, expectedErr
```